src/modules/rlm_couchbase/rlm_couchbase.c

```diff
@@ -452,7 +452,7 @@ static int mod_instantiate(void *instance, CONF_SECTION *conf)
 		bool sep = false;
 
 		len = talloc_array_length(inst->server_raw);
-		server = p = talloc_array(inst, char, len);
+		server = p = talloc_array(inst, char, len+1);
```

Comment:
That's not the correct fix.  talloc_array_length(inst->server_raw) will return a length including the '\0' byte.  Which is the actual issue here, as it causes the loop advance p past the end of the new string.

The correct fix is:

```
len = talloc_array_length(inst->server_raw) - 1;
server = p = talloc_array(inst, char, len + 1);
```

With your fix, the ``server`` ends up with two terminating null bytes.

Reply:
Done @arr2036